challenge_3/cpp/blakebarnes00/src/main.cpp

```diff
@@ -0,0 +1,54 @@
+#include <iostream>
+#include <vector>
+#include <algorithm>
+
+void print_array(std::vector<int> array)
+{
+  for(int i = 0; i < array.size(); i++)
```

Comment:
Think of it this way. Since this is iterating over a stl container, the compiler transforms the ranged based for loop into 
```
for (std::vector<int>::iterator it = array.begin(); it != array.end(); it++) {
  int i = *it;
```

If you need to iterate in reverse, you would need to use a traditional loop. For vector you could use a reverse iterator, instead of the forward iterator in the loop I have above. You could call std::reverse on the container to reverse it and then just loop forward over it. Or you could use the traditional index based approach.